logicalplan/distribute.go

```diff
@@ -148,20 +152,52 @@ func newRemoteAggregation(rootAggregation *parser.AggregateExpr, engines []api.R
 	return &remoteAggregation
 }
 
-func (m DistributedExecutionOptimizer) makeSubQueries(current *parser.Expr, engines []api.RemoteEngine) Deduplicate {
-	remoteQueries := make(RemoteExecutions, len(engines))
-	for i := 0; i < len(engines); i++ {
-		remoteQueries[i] = RemoteExecution{
-			Engine: engines[i],
-			Query:  (*current).String(),
+func (m DistributedExecutionOptimizer) makeSubQueries(expr *parser.Expr, engines []api.RemoteEngine, opts *Opts) Deduplicate {
```

Comment:
Let's add a detailed comment for this method, on how it helps create the subquery?

Reply:
Good point, comment should be there now. 

I also renamed the function to better indicate what it's doing.